sqlglot/transforms.py

```diff
@@ -161,6 +161,67 @@ def unnest_to_explode(expression: exp.Expression) -> exp.Expression:
     return expression
 
 
+def explode_to_unnest(expression: exp.Expression) -> exp.Expression:
+    """Convert explode/posexplode into unnest (used in hive -> presto)."""
+    if isinstance(expression, exp.Select):
+        from sqlglot.optimizer.scope import build_scope
+
+        taken_select_names = set(expression.named_selects)
+        taken_source_names = set(build_scope(expression).selected_sources)
+
+        for select in expression.selects:
+            to_replace = select
+
+            pos_alias = ""
+            explode_alias = ""
+
+            if isinstance(select, exp.Alias):
+                explode_alias = select.alias
+                select = select.this
+            elif isinstance(select, exp.Aliases):
+                pos_alias = select.aliases[0].name
+                explode_alias = select.aliases[1].name
+                select = select.this
+
+            if isinstance(select, (exp.Explode, exp.Posexplode)):
+                is_posexplode = isinstance(select, exp.Posexplode)
+
+                explode_arg = select.this
+                unnest = exp.Unnest(expressions=[explode_arg.copy()], ordinality=is_posexplode)
+
+                # This ensures that we won't use [POS]EXPLODE's argument as a new selection
+                if isinstance(explode_arg, exp.Column):
+                    taken_select_names.add(explode_arg.output_name)
+
+                unnest_source_alias = find_new_name(taken_source_names, "_u")
+                taken_source_names.add(unnest_source_alias)
+
+                if not explode_alias:
+                    explode_alias = find_new_name(taken_select_names, "_c")
+                    taken_select_names.add(explode_alias)
+
+                    if is_posexplode:
+                        pos_alias = find_new_name(taken_select_names, "_c")
```

Comment:
Should this be more descriptive? It's a position column, so it being the same name as the unnested column (both are "_c") seems a bit weird to me. Could it be "_col" and "_pos", or "_c" and "_i" or something?

Reply:
Actually, since explode and posexplode default to pos and col in Spark, imo it would be better if we defaulted to that in the translated queries. (It would be better for our users) Otherwise their queries will break if they depend on the names of the columns (as many do). What do you think?

Reply:
> Should this be more descriptive? It's a position column, so it being the same name as the unnested column (both are "_c") seems a bit weird to me. Could it be "_col" and "_pos", or "_c" and "_i" or something?

This seems reasonable to me and it's not a big deal to apply this change. Thanks for the suggestion.

> Actually, since explode and posexplode default to pos and col in Spark, imo it would be better if we defaulted to that in the translated queries. (It would be better for our users) Otherwise their queries will break if they depend on the names of the columns (as many do). What do you think?

I'm not sure about this one. I was skeptical too, but Spark's approach is not ideal either because we may end up with duplicate column names and hence ambiguity if we just hardcode "col" and "pos". For example:

```
spark-sql> with t(pos) as (select 1) SELECT pos, posexplode(sequence(2, 3)) from t;
pos	pos	col
1	0	2
1	1	3
Time taken: 0.529 seconds, Fetched 2 row(s)
```

If we preserve spark's naming scheme, it will result in an invalid query for presto / trino:

```
trino> WITH t(pos) AS (SELECT 1) SELECT pos, pos, _c_2 FROM t CROSS JOIN UNNEST(SEQUENCE(2, 3)) WITH ORDINALITY AS _u(_c_2, pos);
Query 20230428_200142_00026_tqcwb failed: line 1:34: Column 'pos' is ambiguous
WITH t(pos) AS (SELECT 1) SELECT pos, pos, _c_2 FROM t CROSS JOIN UNNEST(SEQUENCE(2, 3)) WITH ORDINALITY AS _u(_c_2, pos)
```

Here, I just renamed `_c` to `pos` to demonstrate the point, but you get the idea. Maybe a better alternative would be to try and use the `pos`, `col` names, unless there's an ambiguity, in which case we'd provide a warning of some sort and continue with the auto-generated names. Then, we'd get a valid query but downstream references to `pos` might break.

Reply:
Ah yes, I meant to default to col and pos but to add the postfix if those are taken.

Reply:
Posexplode doesn't allow for an alis since it results in two columns, so in that case the user is at the mercy of those names

Reply:
Sorry, it's actually possible by providing a tuple of names! Like we do here for unnest with ordinality, I suppose! So
`select posexplode(col) as (c, p)` works.

https://stackoverflow.com/a/54310860/3600147

Reply:
> Ah yes, I meant to default to col and pos but to add the postfix if those are taken. Added suggested commits for it below. This would be very helpful to us, so hoping you choose that over the ones with underscore!

Ah, great! Sure thing, will make the fix.

Reply:
Done.

Reply:
Thank you!